Source/FakeItEasy.Tests/PriorityTests.cs

```diff
@@ -0,0 +1,214 @@
+        {
+            left.Equals(right).Should().BeFalse();
+        }
+
+        [TestCase(null)]
+        [TestCase(0)]
+        [TestCase((byte)0)]
+        [TestCase("zero")]
+        public void Equals_object_should_return_false_when_other_value_is_another_type(object other)
+        {
+            ShortPriority(0).Equals(other).Should().BeFalse();
+        }
+
+        [TestCaseSource("VariousPriorityValues")]
+        public void GetHashCode_should_return_same_hash_code_as_wrapped_value(short value)
+        {
+            ShortPriority(value).GetHashCode().Should().Be(value.GetHashCode());
+        }
+
+        [TestCaseSource("EqualCases")]
+        public void Equality_operator_should_return_true_when_left_value_is_equal_to_right(Priority left, Priority right)
+        {
+            (left == right).Should().BeTrue();
+        }
+
+        [TestCaseSource("UnequalCases")]
+        public void Equality_operator_should_return_false_when_left_value_is_not_equal_to_right(Priority left, Priority right)
+        {
+            (left == right).Should().BeFalse();
+        }
+
+        [TestCaseSource("EqualCases")]
+        public void Inequality_operator_should_return_false_when_left_value_is_equal_to_right(Priority left, Priority right)
+        {
+            (left != right).Should().BeFalse();
+        }
+
+        [TestCaseSource("UnequalCases")]
+        public void Inequality_operator_should_return_true_when_left_value_is_not_equal_to_right(Priority left, Priority right)
+        {
+            (left != right).Should().BeTrue();
+        }
+
+        [SuppressMessage("Microsoft.Performance", "CA1811:AvoidUncalledPrivateCode", Justification = "Required for testing.")]
+        private static IEnumerable<Priority[]> UnequalCases()
+        {
+            return LessThanCases.Concat(GreaterThanCases());
+        }
+
+        [SuppressMessage("Microsoft.Performance", "CA1811:AvoidUncalledPrivateCode", Justification = "Required for testing.")]
+        private static IEnumerable<Priority[]> GreaterThanCases()
+        {
+            return LessThanCases.Select(pair => new[] { pair[1], pair[0] });
+        }
+
+        [SuppressMessage("Microsoft.Performance", "CA1811:AvoidUncalledPrivateCode", Justification = "Required for testing.")]
+        private static IEnumerable<Priority[]> LessThanOrEqualToCases()
+        {
+            return LessThanCases.Concat(EqualCases);
+        }
+
+        [SuppressMessage("Microsoft.Performance", "CA1811:AvoidUncalledPrivateCode", Justification = "Required for testing.")]
+        private static IEnumerable<Priority[]> GreaterThanOrEqualToCases()
+        {
+            return GreaterThanCases().Concat(EqualCases);
+        }
+
+        private static Priority BytePriority(byte value)
+        {
+            return new Priority(value);
+        }
+
+        private static Priority ShortPriority(short value)
+        {
+            return new Priority(value);
+        }
+    }
+}

```

Comment:
The coverage looks pretty extensive, but dotCover reports 0%... looks like a bug in dotCover.